climakitae/threshold_tools.py

```diff
@@ -0,0 +1,613 @@
+
+    new_ds = lmoments.to_dataset()
+    new_ds = new_ds.rename({"T2": "lmoments"})
+
+    if multiple_points == True:
+        new_ds = new_ds.unstack("allpoints")
+
+    new_ds.attrs = ams_attributes
+    new_ds.attrs["distribution"] = "{}".format(str(distr))
+
+    return new_ds
+
+
+#####################################################################
+
+
+def get_aicc_stat(ams, multiple_points=True):
+
+    ams_attributes = ams.attrs
+
+    if multiple_points == True:
+        ams = ams.stack(allpoints=["x", "y"]).squeeze().groupby("allpoints")
+
+    def apply_function(ams):
+        def function(ams):
+
+            try:
+                lmoments_gev = ldistr.gev.lmom_fit(ams)
+                aicc_gev = ["gev", lstats.AICc(ams, "gev", lmoments_gev)]
+            except ValueError:
+                aicc_gev = ["gev", np.nan]
+
+            try:
+                lmoments_gum = ldistr.gum.lmom_fit(ams)
+                aicc_gum = ["gumbel", lstats.AICc(ams, "gum", lmoments_gum)]
+            except ValueError:
+                aicc_gum = ["gumbel", np.nan]
+
+            try:
+                lmoments_wei = ldistr.wei.lmom_fit(ams)
+                aicc_wei = ["weibull", lstats.AICc(ams, "wei", lmoments_wei)]
+            except ValueError:
+                aicc_wei = ["weibull", np.nan]
+
+            try:
+                lmoments_pe3 = ldistr.pe3.lmom_fit(ams)
+                aicc_pe3 = ["pearson3", lstats.AICc(ams, "pe3", lmoments_pe3)]
+            except ValueError:
+                aicc_pe3 = ["pearson3", np.nan]
+
+            try:
+                lmoments_gpa = ldistr.gpa.lmom_fit(ams)
+                aicc_gpa = ["genpareto", lstats.AICc(ams, "gpa", lmoments_gpa)]
+            except ValueError:
+                aicc_gpa = ["genpareto", np.nan]
+
+            all_aicc_results = (
+                str(aicc_gev)
+                + str(aicc_gum)
+                + str(aicc_wei)
+                + str(aicc_pe3)
+                + str(aicc_gpa)
+            )
+            all_aicc_results_string = str(all_aicc_results)
+
+            lowest_aicc_value = min(
+                aicc_gev[1], aicc_gum[1], aicc_wei[1], aicc_pe3[1], aicc_gpa[1]
+            )
+
+            if lowest_aicc_value == aicc_gev[1]:
+                lowest_aicc_distr = aicc_gev[0]
+            elif lowest_aicc_value == aicc_gum[1]:
+                lowest_aicc_distr = aicc_gum[0]
+            elif lowest_aicc_value == aicc_wei[1]:
+                lowest_aicc_distr = aicc_wei[0]
+            elif lowest_aicc_value == aicc_pe3[1]:
+                lowest_aicc_distr = aicc_pe3[0]
+            elif lowest_aicc_value == aicc_gpa[1]:
+                lowest_aicc_distr = aicc_gpa[0]
+
+            return all_aicc_results, lowest_aicc_distr, lowest_aicc_value
+
+        return xr.apply_ufunc(
+            function,
+            ams,
+            input_core_dims=[["time"]],
+            exclude_dims=set(("time",)),
+            output_core_dims=[[], [], []],
+        )
```

Comment:
You may want to try adding the argument dask='allowed'... which I expect should work, or else the 'parallelize' option. Doing the step of loading 'ams' in the notebook is not optimal -- takes memory and time unnecessarily.
https://xarray.pydata.org/en/stable/generated/xarray.apply_ufunc.html